scripts/check_e2e_tests_are_captured_in_ci_test.py

```diff
@@ -0,0 +1,241 @@
+    """
+
+    def _mock_read_protractor_conf_file(self):
+        protractor_config_file = python_utils.open_file(
+            os.path.join(
+                DUMMY_CONF_FILES, 'dummy_protractor.conf.js'), 'r').read()
+        return protractor_config_file
+
+    def _mock_read_travis_yml_file(self):
+        travis_ci_file = python_utils.open_file(
+            os.path.join(
+                DUMMY_CONF_FILES, '.dummy_travis.yml'), 'r').read()
+        travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
+        return travis_ci_dict
+
+    def _mock_read_protractor_conf_file_fail(self):
+        protractor_config_file = python_utils.open_file(
+            os.path.join(
+                DUMMY_CONF_FILES, 'protractor_has_extra_test.conf.js'),
+                'r').read()
+        return protractor_config_file
+
+    def _mock_read_travis_yml_file_fail(self):
+        travis_ci_file = python_utils.open_file(
+            os.path.join(
+                DUMMY_CONF_FILES, '.travis_has_extra_test.yml'), 'r').read()
+        travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
+        return travis_ci_dict
+
+    def _return_empty_list(self):
+        return []
+
+    def test_read_protractor_file(self):
+        expected_protractor_config_file = python_utils.open_file(
+            os.path.join(
+                os.getcwd(), 'core', 'tests', 'protractor.conf.js'), 'r').read()
+        actual_protractor_config_file = (
+            check_e2e_tests_are_captured_in_ci.read_protractor_conf_file())
+        self.assertEqual(expected_protractor_config_file,
+                         actual_protractor_config_file)
+
+    def test_read_travis_ci_file(self):
+        travis_ci_file = python_utils.open_file(
+            os.path.join(os.getcwd(), '.travis.yml'), 'r').read()
+        expected_travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
+        actual_travis_ci_dict = (
+            check_e2e_tests_are_captured_in_ci.read_and_parse_travis_yml_file())
+        self.assertEqual(expected_travis_ci_dict, actual_travis_ci_dict)
+
+    def test_dummy_read_protractor_file(self):
+        expected_protractor_config_file = (python_utils.open_file(
+            os.path.join(
+                DUMMY_CONF_FILES, 'dummy_protractor.conf.js'), 'r').read())
+        dummy_path = self.swap(
+            check_e2e_tests_are_captured_in_ci, 'read_protractor_conf_file',
+            self._mock_read_protractor_conf_file)
+        with dummy_path:
+            actual_protractor_config_file = (
+                check_e2e_tests_are_captured_in_ci.read_protractor_conf_file())
+        self.assertEqual(expected_protractor_config_file,
+                         actual_protractor_config_file)
+
+    def test_dummy_read_travis_ci_file(self):
+        travis_ci_file = python_utils.open_file(
+            os.path.join(
+                DUMMY_CONF_FILES, '.dummy_travis.yml'), 'r').read()
+        expected_travis_ci_dict = utils.dict_from_yaml(travis_ci_file)
```

Comment:
Ditto -- hardcode it. Generally there shouldn't be logic in the test for calculating the expected values.

Reply:
Done